### ISSUE TITLE
Disabled attribute in an input element is failing build compilation

### ISSUE THREAD

**Issue by bh3605** [angular/angular#43626]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

Yes

### Description

Related: https://github.com/angular/components/issues/17495

Updated an app from angular 10 to angular 12.2 to find the lovely disabled attribute is causing my build to fail.

Example:
html

    <form [formGroup]="myForm">
      <input type="text" disabled formControlName="mo">
    </form>

component class

    export class AppComponent {
      constructor(private fb: FormBuilder) {

      }
      myForm: FormGroup = this.fb.group({
        mo: new FormControl('')
      });
    }

Now I could specify `[disabled]="true"` and it would compile then, but I really shouldn't have to given `disabled` by itself is valid.

### Please provide the exception or error you saw

```true
Error: src/app/app.component.html:2:61 - error TS2322: Type 'string' is not assignable to type 'boolean'.

   <input type="text" disabled formControlName="mo">
```

### Please provide the environment you discovered this bug in

```true
"@angular/animations": "~12.2.0",
    "@angular/cdk": "^12.2.7",
    "@angular/common": "~12.2.0",
    "@angular/compiler": "~12.2.0",
    "@angular/core": "~12.2.0",
    "@angular/forms": "~12.2.0",
    "@angular/platform-browser": "~12.2.0",
    "@angular/platform-browser-dynamic": "~12.2.0",
    "@angular/router": "~12.2.0",
"rxjs": "~6.6.0",
```

### Anything else?

Adding tsconfig.json file for extra clarity
/* To learn more about this file see: https://angular.io/config/tsconfig. */
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "forceConsistentCasingInFileNames": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "sourceMap": true,
    "declaration": false,
    "downlevelIteration": true,
    "experimentalDecorators": true,
    "moduleResolution": "node",
    "importHelpers": true,
    "target": "es2017",
    "module": "es2020",
    "lib": [
      "es2018",
      "dom"
    ]
  },
  "angularCompilerOptions": {
    "enableI18nLegacyMessageIdFormat": false,
    "strictInjectionParameters": true,
    "strictInputAccessModifiers": true,
    "strictTemplates": true
  }
}

**Comment by JoostK** [comment]
This is working as intended but I can understand that this causes friction.

You can configure `"strictAttributeTypes": true` to disable this error.

**Comment by JoostK** [comment]
By the way; you should not be using `disabled` when using reactive forms as the disabled state is controlled by the form control.

Do you get this warning in development mode?

```
  It looks like you're using the disabled attribute with a reactive form directive. If you set disabled to true
  when you set up this control in your component class, the disabled attribute will actually be set in the DOM for
  you. We recommend using this approach to avoid 'changed after checked' errors.

  Example:
  form = new FormGroup({
    first: new FormControl({value: 'Nancy', disabled: true}, Validators.required),
    last: new FormControl('Drew', Validators.required)
  });
```

**Comment by bh3605** [comment]
@JoostK Yes, I do see the message you mention pop up in the console. I'm migrating this app to use single-spa and updating the project version. I'll make a note of this. Thanks.
